agent/consul/rate/handler.go

```diff
@@ -172,9 +178,15 @@ func (h *Handler) Allow(op Operation) error {
 }
 
 func (h *Handler) UpdateConfig(cfg HandlerConfig) {
+	existingCfg := h.cfg.Load()
 	h.cfg.Store(&cfg)
-	h.limiter.UpdateConfig(cfg.GlobalWriteConfig, globalWrite)
-	h.limiter.UpdateConfig(cfg.GlobalReadConfig, globalRead)
+	if !reflect.DeepEqual(existingCfg.GlobalWriteConfig, cfg.GlobalWriteConfig) {
+		h.limiter.UpdateConfig(cfg.GlobalWriteConfig, globalWrite)
+	}
+	if !reflect.DeepEqual(existingCfg.GlobalReadConfig, cfg.GlobalReadConfig) {
+		h.limiter.UpdateConfig(cfg.GlobalReadConfig, globalRead)
+
+	}
```

Comment:
I think we should add a TODO to log some message when we don't apply the config here.

Reply:
@dhiaayachi Thank you for calling this out.  What do you think of something like this?  (It's my first foray into adding logging and wasn't sure if there was another way todo this.  Also, is warning the right level?)

https://github.com/hashicorp/consul/pull/15805/commits/33c36d25241b368f64883d4b784a2b70dab9c502